### ISSUE TITLE
[TKLDEV] Missing path in common for fab-apply-patch

### ISSUE THREAD

**Issue by jstruebel** [turnkeylinux/tracker#856]
I recently upgraded my TKLDEV appliance to the latest packages and now I'm running into an error from what I think is the new fab-apply-patch that @Dude4Linux added. The specific error that I get is:

```
# apply the common patches
fab-apply-patch /turnkey/fab/common/patches/turnkey.d build/root.patched
error: does not exist: /turnkey/fab/common/patches/turnkey.d
/usr/share/fab/product.mk:526: recipe for target 'build/stamps/root.patched' failed
make: *** [build/stamps/root.patched] Error 1
```

I pulled the latest changes from https://github.com/turnkeylinux/common.git thinking that might fix it, but I still don't have a patches directory. Since this error terminates the build, I'm unable to continue building any appliances. If I add an empty directory at /turnkey/fab/common/patches/turnkey.d then the appliance build completes. I propose either adding the required directory to the common repo or changing the fab-apply-patch or makefiles to continue if the patches directory is missing so that end users of tkldev don't need to create the directory manually.

If there is already a way to deal with this case I'd be happy to learn about it.

**Comment by JedMeister** [comment]
Thanks @jstruebel - I actually noticed exactly the same thing late last week and fixed it (by creating the directory). I did push back to GH, but forgot to do a PR and merge. Thanks for the prompt!

And hopefully you should also notice that the incremental build targets should be working properly again now (with the update to `fab`). I think I've inadvertently been telling people to update `deck`, but it was actually the `fab` package which fixes it!

Anyway, I've just fixed this issue in the master repo. Hopefully a `git pull origin master` (or whatever you are using for TurnKey's GH repo if not `origin`) you should be all go again now.

**Comment by JedMeister** [comment]
PS in some respects, just making that section of code ignore the missing dir may actually be preferable. But I've just added the dir as then people may notice the additional functionality. FWIW I preferred the option of leaving fab as it was and including (and updating) buildtasks. But Liraz merged so we'll run with that...

**Comment by Dude4Linux** [comment]
@jstruebel @JedMeister - Sorry guys, my bad.  Git has a bad habit of ignoring empty directories and I guess I failed to include an anchor file.  Apparently I never tested the case where the directory was missing. Thanks for catching and fixing the problem.

**Comment by JedMeister** [comment]
Hey @Dude4Linux No problems mate. Liraz probably should have tested himself before merging (or made sure someone else tested it further).

No major harm done because at this point, the only users who would notice are ones like Jonathan who have updated. It will be included by default in TKLDev 14.2.
